src/main/java/com/google/api/codegen/discovery/DefaultString.java

```diff
@@ -94,14 +94,22 @@ static String forPattern(String pattern, String placeholderFormat) {
     }
 
     StringBuilder ret = new StringBuilder();
-    for (int i = 0; i < elems.size(); i += 2) {
-      String literal = elems.get(i).getLiteral();
-      String placeholder = Inflector.singularize(literal);
-      placeholder = LanguageUtil.lowerCamelToLowerUnderscore(placeholder).replace('_', '-');
-      ret.append('/')
-          .append(literal)
-          .append("/")
-          .append(String.format(placeholderFormat, placeholder));
+    for (int i = 0; i < elems.size(); i++) {
```

Comment:
Given the revised specification, it seems more appropriate to keep the looping by two, and simply add an additional check at the end for an extra literal.